### DISCUSSION TITLE
Exportfeed

### DISCUSSION THREAD

**Discussion by ghost** [atrocore/atrocore#674] [Q&A]
Hay guys, 

two questions about the Exportfeed:

1. How to properly use the option to define a specific column name based on the database? 
![image](https://user-images.githubusercontent.com/127828659/225888807-a1ca3c90-a9f8-4e17-9a66-a48893908f5f.png)
I want to use the code as column header but using the syntax as described in the tooltip, it basically just says {{code}} instead of replacing it with the actual value of the column
![image](https://user-images.githubusercontent.com/127828659/225889151-1ee1173a-8787-4927-aa1c-1f34cbffad0e.png)

2. Is it somehow possible to export a unit attribute into two columns based on value + unit? As you can see in the screenshot above it combines "300 g" into one column, but i would like to have the "300" and the "g" in two separate columns. Basically as its actually saved in the database:
![image](https://user-images.githubusercontent.com/127828659/225889690-add65e48-dab3-4347-95fd-fb1cab9b6ca2.png)

Thank you very much and have a nice weekend!

Edit:
3. Not sure if this is a bug or intention, but if an attribute value is "0", its empty in the export 
![image](https://user-images.githubusercontent.com/127828659/225905486-382262b2-d66d-4246-aee6-5bd07aea0c42.png)
![image](https://user-images.githubusercontent.com/127828659/225905551-46a27ccb-0939-40ad-a0ec-c323e31db81f.png)

In my opinion it definitely should export the zero as it its a possible value as well, especially related to a unit

**Comment by atrolex** [comment]
1. Generally, for each column in the export file you need to create a rule in the "configurator" panel. If you set the "Field name" (Spaltenbezeichnung) as "Name", and not as "Custom Name", the column will be named after field name or attribute name automatically.
By choosing "Name / Locale Suffix" column names will be set in according language.
I think {{code}} would not work for you, because technically product attribute values are exported from this entity:
https://demo.atropim.com/#Admin/fieldManager/scope=ProductAttributeValue
which does not have a field named "code", you have here only attribute ID and Name.

2. Create 2 rules in configurator, in one set Maske = {{value}}, in another Maske = {{unit}}

3. As for me it sounds like a bug. We will check it, thx!

**Reply by ghost** [reply]
Hi @atrolex , thank you for the quick answer

Regarding 1)
I just tried to use **{{attribute_id}}** (db field name) as well as **{{attributeId}}** which is definitely available in the scope you linked; but the export still does not transforms it into the value
![image](https://user-images.githubusercontent.com/127828659/226542141-4b1cc839-9f40-43ae-bbea-e404f8ea5551.png)

Can you please check it again? Thank you!

**Reply by atrolex** [reply]
I think, this functionality was created only for the fields, not for the attributes, i think, that is why it is not working. I think you expect to have some dynamic logic. It does not work like that, for one column you need to have one rule in configurator. There is only one exception, multilink, for multilink records, you can use the checkbox "Separate" in this case columns like xxx_1,xxx_2, xxx_3 will be automatically created.

**Reply by ghost** [reply]
Im not sure i do understand the tooltip correctly then.. even if i use this "mask" logic on a field and not attribute its not working:

![image](https://user-images.githubusercontent.com/127828659/226556132-65c080a3-59bd-4687-96f8-d280d2b151d8.png)
![image](https://user-images.githubusercontent.com/127828659/226556225-18ec07a9-9dee-4691-bf4b-b9e86e4e2beb.png)

From my understand it should replace the column name based on the database field value using the syntax - or am i totally wrong? :D

**Reply by atrolex** [reply]
try {{createdAt}}

**Reply by ghost** [reply]
For now, i simply change it on the database level:

UPDATE `export_configurator_item` SET `column` = (SELECT code from attribute where id = export_configurator_item.attribute_id), column_type = 'custom' where `export_feed_id` = 'xxx' and deleted = 0 and `column` = null

This changes the exports column names to "code" from attributes, but can be replaced by any db field from the sub query

![image](https://user-images.githubusercontent.com/127828659/226560562-ce1e39a4-287f-480a-9861-995c6a1889be.png)

Would be cool to have this as feature by default as the attribute name can be the same (as its in my case) just within different attribute groups and after the export no one knows what it belongs too based on the name

**Reply by ghost** [reply]
> 2\. Create 2 rules in configurator, in one set Maske = {{value}}, in another Maske = {{unit}}

This is also not possible, as i cant use the same attribute and add it a second time to differ between masks

**Reply by atrolex** [reply]
2. worked for me:
![image](https://user-images.githubusercontent.com/33658481/226665065-ec53d0f6-d8ff-4a6d-8e4b-0e695875293d.png)

**Reply by atrolex** [reply]
As for your previous post:
In our world you need to create a rule for the column, and thus, you already know which attribute you have selected, so you can change to "custom name" and set any name, also the same name as your code is. Maybe it will take a little bit more time, but usually such configurations are not done/changed frequently. Main thing you get the same result :) By choosing attribute, in the same time you can copy the "code", and then insert it in the custom name field.

**Reply by atrolex** [reply]
Additionally, you can export the data directly from the entity Attribute Value – in this case you will have one column for the attribute name, and another for the attribute value. In that case everything will be generated dynamically, top-to-bottom, not left-to-right, as via products. You can also directly select "Code" to be used instead of attribute name.

**Reply by ghost** [reply]
> 2. worked for me:
>    ![image](https://user-images.githubusercontent.com/33658481/226665065-ec53d0f6-d8ff-4a6d-8e4b-0e695875293d.png)

Oh ok, found the 'reason'
If you pick the attribute via the "plus" 
![image](https://user-images.githubusercontent.com/127828659/226820763-6b694e24-0ed3-417b-aca6-44cc9abbdd33.png)
it works to add it a second time 

but not using the other option via the "arrow down" -> pick attributes
![image](https://user-images.githubusercontent.com/127828659/226820919-37d70a47-7d17-4f92-ae1a-5bc651c7fb99.png)

I.e. it works _only_ via this interface:
![image](https://user-images.githubusercontent.com/127828659/226820635-2917c009-7351-4c07-8260-73d5673d0159.png)

Would it be possible to add it to the multi select as well?

**Reply by ghost** [reply]
> As for your previous post: In our world you need to create a rule for the column, and thus, you already know which attribute you have selected, so you can change to "custom name" and set any name, also the same name as your code is. Maybe it will take a little bit more time, but usually such configurations are not done/changed frequently. Main thing you get the same result :) By choosing attribute, in the same time you can copy the "code", and then insert it in the custom name field.

Well, thats also true; makes sense, thank you
